src/Microsoft.AspNetCore.Blazor.Server/Circuits/RemoteUriHelper.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Blazor.Services;
+
+namespace Microsoft.AspNetCore.Blazor.Server.Circuits
+{
+    // TODO: Stop duplicating all this implementation.
+    // Maybe have a UriHelperBase abstract base class
+    internal class RemoteUriHelper : IUriHelper
+    {
+        private const string _functionPrefix = "Blazor._internal.uriHelper.";
+
+        private readonly IJSRuntimeAccessor _jsRuntimeAccessor;
+        private string _uriAbsolute;
+
+        // These two are always kept in sync. We store both representations to
+        // avoid having to convert between them on demand.
+        private Uri _baseUriWithTrailingSlash;
+        private string _baseUriStringWithTrailingSlash;
+
+        public RemoteUriHelper(IJSRuntimeAccessor jsRuntimeAccessor)
+        {
+            _jsRuntimeAccessor = jsRuntimeAccessor ?? throw new ArgumentNullException(nameof(jsRuntimeAccessor));
+        }
+
+        public void Initialize(string uriAbsolute, string baseUriAbsolute)
+        {
+            _baseUriStringWithTrailingSlash = ToBaseUri(baseUriAbsolute);
+            _baseUriWithTrailingSlash = new Uri(_baseUriStringWithTrailingSlash);
+
+            _uriAbsolute = uriAbsolute;
+
+            _jsRuntimeAccessor.JSRuntime.InvokeAsync<object>(_functionPrefix + "enableNavigationInterception");
+        }
```

Comment:
LMK what you think of these changes. In general you can't allow services to have a dependency on data without some kind of lifecycle. I opted to do the simplest thing which is to have a lifecycle with 2 steps 🤣

Reply:
The way it passes in `uriAbsolute` and `baseUriAbsolute` looks great.

As per comment above, I'm wondering if we could eliminate the `InvokeAsync ... enableNavigationInterception` and put that logic directly in `Boot.WebAssembly.ts` since it's not going to be optional.

Reply:
Update: I mean `Boot.Server.ts`.